src/compiler/utilities.ts

```diff
@@ -4298,6 +4298,11 @@ namespace ts {
         return declaration.name || nameForNamelessJSDocTypedef(declaration);
     }
 
+    /** @internal */
+    export function isNamedDeclaration(node: Node): node is NamedDeclaration & { name: DeclarationName } {

```

Comment:
Mmm, this implementation doesn't seem to actually check that the name is a `DeclarationName`.

Reply:
Added an assertion -- a `name` property should always be a `DeclarationName`.